wrappers/services_gen_test.go

```diff
@@ -201,3 +202,29 @@ apps:
 
 	c.Assert(wrapperText, Equals, expectedDbusService)
 }
+
+func (s *servicesWrapperGenSuite) TestGenOneshotServiceFile(c *C) {
+
+	yamlText := `
+name: snap
+version: 1.0
+apps:
+    app:
+        command: bin/start
+        stop-command: bin/stop
+        reload-command: bin/reload
+        post-stop-command: bin/stop --post
+        stop-timeout: 10s
+        daemon: oneshot
+`
+
+	info, err := snap.InfoFromSnapYaml([]byte(yamlText))
```

Comment:
Sorry, I meant MockInfo :-) They differ subtly but I think all you need is MockInfo